tests/parser/base.py

```diff
@@ -150,7 +164,9 @@ def assert_nm_regenerate(self, file_contents_map):
         argv = [exe_generate, '--root-dir', self.workdir.name]
         p = subprocess.Popen(argv, stdout=subprocess.PIPE,
                              stderr=subprocess.PIPE, universal_newlines=True)
+        returncode = p.wait(5)
```

Comment:
praise: Good catch!

This seem like the sort of thing that could cause serious headaches!